modules/ocf_backups/files/upload-to-box

```diff
@@ -0,0 +1,444 @@
+# token, so it is stored in a local file (/opt/share/backupsbox-refresh-token)
+# and overwritten each time a new one is returned (when the old one is used).
+# Access tokens only last for an hour before expiring, so they are only stored
+# in this program's memory and are generated using the refresh token.
+
+
+class BoxUpload:
+
+    def __init__(self, filename, curl_auth):
+        self.filename = filename
+        self.auth = curl_auth
+        self.percentage = 0.0
+        self.bytes_uploaded = 0
+
+    def start(self, conn):
+        """Set curl params for the file to upload"""
+        file_path = ARCHIVE + '/' + self.filename
+        conn.upload = self
+
+        # PROGRESSFUNCTION is deprecated in pycurl >= 7.32.0, so this
+        # should be changed to conn.XFERINFOFUNCTION when we upgrade to stretch
+        conn.setopt(conn.PROGRESSFUNCTION, self.progress)
+        conn.setopt(conn.NOPROGRESS, False)
+
+        conn.file_bytes = os.path.getsize(file_path)
+        conn.fp = open(file_path, 'rb')
+
+        conn.setopt(conn.INFILESIZE, conn.file_bytes)
+        conn.setopt(conn.READFUNCTION, conn.fp.read)
+        conn.setopt(conn.UPLOAD, True)
+        conn.setopt(conn.URL, '{}{}/{}'.format(BOX_DAV, FOLDER, self.filename))
+        conn.setopt(conn.USERPWD, self.auth)
+
+    def progress(self, download_t, download_d, upload_t, upload_d):
+        """Store the upload progress percentage for periodic printing.
+
+        Passed to pycurl as a progress function for each upload
+        """
+        self.bytes_uploaded = upload_d
+        if upload_t > 0:
+            self.percentage = 100 * (upload_d / upload_t)
+
+    def get_progress(self, terminal_width):
+        """Get a nice progress bar including the file and percent complete
+
+        The progress bar has a dynamic width, so it must be constructed by
+        figuring out the correct width based on other data shown
+        """
+        first_half = '{} ['.format(self.filename)
+        second_half = '] {:.2f}%'.format(self.percentage)
+        width = terminal_width - len(first_half) - len(second_half)
+        filled = int(width * self.percentage / 100)
+
+        return '{}{}{}{}'.format(
+            first_half,
+            filled * '#',
+            (width - filled) * ' ',
+            second_half
+        )
+
+
+def box_api_call(url, data='', headers=[], method=''):
+    """Make an API call to Box.com (or anywhere really)"""
+    conn = pycurl.Curl()
+
+    response_data = BytesIO()
+    conn.setopt(conn.URL, url)
+    conn.setopt(conn.WRITEDATA, response_data)
+    conn.setopt(conn.HTTPHEADER, headers)
+
+    if data:
+        conn.setopt(conn.POSTFIELDS, data)
+
+    if method:
+        conn.setopt(conn.CUSTOMREQUEST, method)
+
+    conn.perform()
+    http_status = conn.getinfo(pycurl.HTTP_CODE)
+    conn.close()
+
+    # Check if HTTP status is in 2XX (success of some sort)
+    if http_status in range(200, 299):
```

Comment:
does this look better as `if 200 <= http_status < 300:`?

that's probably how i'd write it, this seems fine though